geomstats/geometry/hyperbolic.py

```diff
@@ -612,330 +438,46 @@ def squared_norm(self, vector, base_point=None):
-        base_point : array-like, shape=[n_samples, dimension + 1]
-            Point in hyperbolic space.
+        vector : array-like, shape=[n_samples, dimension + 1]
+            Vector on the tangent space of the hyperbolic space at base point.
+        base_point : array-like, shape=[n_samples, dimension + 1], optional
+            Point in hyperbolic space in extrinsic coordinates.
 
         Returns
         -------
-        log : array-like, shape=[n_samples, dimension + 1]
-            Tangent vector at the base point equal to the Riemannian logarithm
-            of point at the base point.
+        sq_norm : array-like, shape=[n_samples, 1]
+            Squared norm of the vector.
         """
-        point = gs.to_ndarray(point, to_ndim=2)
-        base_point = gs.to_ndarray(base_point, to_ndim=2)
-
-        if self.coords_type == 'extrinsic':
-
-            angle = self.dist(base_point, point) / self.scale
-            angle = gs.to_ndarray(angle, to_ndim=1)
-            angle = gs.to_ndarray(angle, to_ndim=2)
-
-            mask_0 = gs.isclose(angle, 0.)
-            mask_else = ~mask_0
-
-            mask_0_float = gs.cast(mask_0, gs.float32)
-            mask_else_float = gs.cast(mask_else, gs.float32)
-
-            coef_1 = gs.zeros_like(angle)
-            coef_2 = gs.zeros_like(angle)
-
-            coef_1 += mask_0_float * (
-                1. + INV_SINH_TAYLOR_COEFFS[1] * angle ** 2
-                + INV_SINH_TAYLOR_COEFFS[3] * angle ** 4
-                + INV_SINH_TAYLOR_COEFFS[5] * angle ** 6
-                + INV_SINH_TAYLOR_COEFFS[7] * angle ** 8)
-            coef_2 += mask_0_float * (
-                1. + INV_TANH_TAYLOR_COEFFS[1] * angle ** 2
-                + INV_TANH_TAYLOR_COEFFS[3] * angle ** 4
-                + INV_TANH_TAYLOR_COEFFS[5] * angle ** 6
-                + INV_TANH_TAYLOR_COEFFS[7] * angle ** 8)
-
-            # This avoids dividing by 0.
-            angle += mask_0_float * 1.
-
-            coef_1 += mask_else_float * (angle / gs.sinh(angle))
-            coef_2 += mask_else_float * (angle / gs.tanh(angle))
-
-            log = (gs.einsum('...i,...j->...j', coef_1, point) -
-                   gs.einsum('...i,...j->...j', coef_2, base_point))
-            return log
-
-        elif self.coords_type == 'ball':
-
-            add_base_point = self.mobius_add(-base_point, point)
-            norm_add = gs.to_ndarray(gs.linalg.norm(
-                add_base_point, axis=-1), 2, -1)
-            norm_add = gs.repeat(norm_add, base_point.shape[-1], -1)
-            norm_base_point = gs.to_ndarray(gs.linalg.norm(
-                base_point, axis=-1), 2, -1)
-            norm_base_point = gs.repeat(norm_base_point,
-                                        base_point.shape[-1], -1)
-
-            log = (1 - norm_base_point**2) * gs.arctanh(norm_add)\
-                * (add_base_point / norm_add)
+        return super().squared_norm(vector, base_point=base_point)
```

Comment:
Will this work? Since super() refers to EmbeddingManifold but no embedding_manifold is provided?

Reply:
I Think here super() refer to RiemannianMetric. However, it only works because inner_product_matrix is override in its child.

Reply:
Right!

Reply:
But in this case you don't need to override it here: the RiemannianMetric method will be called by default.

Reply:
Got it: you actually need the override it because of the scale parameter, right? Looks good then.